src/filter/http_context.rs

```diff
@@ -152,33 +133,14 @@ impl Filter {
                         Some(key) => key.to_owned(),
                     };
 
-                    let attribute_path = tokenize_with_escaping(&selector_item.selector, '.', '\\');
-                    // convert a Vec<String> to Vec<&str>
-                    // attribute_path.iter().map(AsRef::as_ref).collect()
-                    let value = match self
-                        .get_property(attribute_path.iter().map(AsRef::as_ref).collect())
-                    {
-                        None => {
-                            debug!(
-                                "[context_id: {}]: selector not found: {}",
-                                self.context_id, selector_item.selector
-                            );
-                            match &selector_item.default {
-                                None => return None, // skipping the entire descriptor
-                                Some(default_value) => default_value.clone(),
-                            }
-                        }
-                        // TODO(eastizle): not all fields are strings
-                        // https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes
-                        Some(attribute_bytes) => match String::from_utf8(attribute_bytes) {
-                            Err(e) => {
-                                debug!(
-                                "[context_id: {}]: failed to parse selector value: {}, error: {}",
-                                self.context_id, selector_item.selector, e);
-                                return None;
-                            }
-                            Ok(attribute_value) => attribute_value,
-                        },
+                    let typed_property = match self.get_typed_property(&selector_item.selector) {
+                        Some(typed_property) => typed_property,
+                        None => return None,
```

Comment:
where is `&selector_item.default` being used when the typed property is not found?

Reply:
Not there right now, as it itself introduces a rather non neglectable refactoring of this PR: tl;dr it needs type inference and coercing from the now not `String` value, but a Value literal.